src/tokenizer/ngram_tokenizer.rs

```diff
@@ -0,0 +1,121 @@
+use super::{Token, TokenStream, Tokenizer};
+
+///Tokenize the text by splitting words into ngrams of the given size
+#[derive(Clone)]
```

Comment:
We need more info about what it does.

Especially what are the resulting token's `position`.

Maybe even move the example as a doctest.